base/skipoftype.jl

```diff
@@ -0,0 +1,145 @@
+struct SkipOfType{T, A}
+    x::A
+end
+
+"""
+    skipoftype(T, itr)
+Return an iterator over the elements in `itr` skipping values of type T.
+Use [`collect`](@ref) to obtain an `Array` containing the non-`T` values in
+`itr`. Note that even if `itr` is a multidimensional array, the result will always
+be a `Vector` since it is not possible to remove nothings while preserving dimensions
+of the input.
+# Examples
+```jldoctest
+julia> sum(skipoftype(Nothing, [1, nothing, 2]))
+3
+julia> collect(skipoftype(Missing, [1, missing, 2]))
+2-element Array{Int64,1}:
+ 1
+ 2
+julia> collect(skipoftype(Nothing, [1 nothing; 2 nothing]))
+2-element Array{Int64,1}:
+ 1
+ 2
+```
+"""
+skipoftype(::Type{T}, itr::A) where {T, A} = SkipOfType{T, A}(itr)
+skipoftype(::T, itr) where T = skipoftype(T, itr)
+
+IteratorSize(::Type{<:SkipOfType}) = SizeUnknown()
+IteratorEltype(::Type{SkipOfType{T, A}}) where {T, A} = IteratorEltype(A)
+eltype(::Type{SkipOfType{T, A}}) where {T, A} = union_poptype(T, eltype(A))
```

Comment:
`typesubtract`?

Reply:
Changed

Reply:
`typesubtract` is already a function that exists, at `Core.Compiler.typesubtract`

Reply:
Ah I had no idea! I changed the PR to use the existing function.